src/main/java/bio/terra/service/profile/google/GoogleBillingService.java

```diff
@@ -71,18 +103,20 @@ private static CloudBillingClient cloudBillingClient() {
      *
      * The second permission is specific to projects, so we will check for the first permission here.
      *
+     * @param user
      * @param billingAccountId
-     * @return true if the repository can act as a billing account *user* (viewer is not enough), false otherwise
+     * @return true if a user can act as a billing account *user* (viewer is not enough), false otherwise
      */
-    public boolean canAccess(String billingAccountId) {
+    public boolean canAccess(AuthenticatedUserRequest user, String billingAccountId) {
         ResourceName resource = BillingAccountName.of(billingAccountId);
         List<String> permissions = Collections.singletonList("billing.resourceAssociations.create");
         TestIamPermissionsRequest permissionsRequest = TestIamPermissionsRequest.newBuilder()
             .setResource(resource.toString())
             .addAllPermissions(permissions)
             .build();
         try {
-            TestIamPermissionsResponse response = cloudBillingClient().testIamPermissions(permissionsRequest);
+            logger.info("Testing IAM permission on billing account: {}", billingAccountId);
+            TestIamPermissionsResponse response = cloudBillingClient(user).testIamPermissions(permissionsRequest);
```

Comment:
The test that I am trying to execute is `IT_JADEAPIURL=http://localhost:8080 ./gradlew testIntegration --tests *FileTest.fileParallelFailedLoadTest --scan` . It fails here.